### DISCUSSION TITLE
Overriding/skipping a hook

### DISCUSSION THREAD

**Discussion by Stefanhg** [pytest-dev/pytest#11302] [Q&A] [answered]
Hello, 

I got a defined pytest_runtest_setup in my conftest that sets up my target for general testing but for certain testcases i either want a different setup method or just want to skip it as the setup is not needed. 
Ideally I think it would be to just make the setup as a fixture as then i got proper control of when it was executed but I am still curious..

Is there a way to override or skip the defined pytest_runtest_setup in my conftest?

**Comment by RonnyPfannschmidt** [comment]
Hooks cannot be overridden/skipped easily

Id recommend using the use fixtures marker plus a fixture if you want to have better control over what Tests use a fixtures

I also recommend to outline the concrete use case in more detail to ensure more to the point advice

**Reply by Stefanhg** [reply]
Well it was worth a shot. 

Yeah fixtures is just the way to go, I am just not a fan of having to repetitively add the fixture to each testcase. 

I was also considering it, I just could not provide a good example of how to use it. In my case these testcases would be in 2 separate files.

**Reply by RonnyPfannschmidt** [reply, marked as answer]
if one file needs the fixture, and the other file now, jsut make a usefixtures mark for the second file and/or have a private autouse fixture in one of the files

**Reply by Stefanhg** [reply]
Oh yeah that's right, I can just do that!
That I think I can manage.
